conda/mpich/conda_build_config.yaml

```
@@ -14,7 +14,7 @@ moose_libmesh_vtk:
   - moose-libmesh-vtk 9.1.0 build_1
 
 moose_petsc:
-  - moose-petsc 3.15.1 build_4
+  - moose-petsc 3.16.1 build_0
```

```suggestion
  - moose-petsc 3.16.3 build_0
```
This was not consistent with the version number in `petsc/meta.yaml`

---

Good catch, thanks